### PR TITLE
CAL#695: Add POST endpoint to accept shared calendar invite

### PR DESCRIPTION
**Pull request by renaudboyer**

### CODE REVIEW

File: lib/CalDAV/SharedCalendar.php [hunk 1]
```diff
@@ -217,6 +217,13 @@ function getInviteStatus() {
 
     }
 
+    function updateInviteStatus($status) {
+
```
Comment (abenchi) [inline lib/CalDAV/SharedCalendar.php]: what about all this newlinessssss

File: lib/JSON/Plugin.php [hunk 2]
```diff
@@ -800,6 +800,35 @@ function isHiddenPrivateEvent($vevent, $node) {
...
+                switch ($jsonData->{'invite-reply'}->{'invitestatus'}) {
+                    case 'accepted': 
+                        $inviteStatus = \Sabre\DAV\Sharing\Plugin::INVITE_ACCEPTED;
+                        break;
+                    case 'noresponse':
+                        $inviteStatus = \ESN\DAV\Sharing\Plugin::INVITE_NORESPONSE;
+                }
+            }
+
+            $node->updateInviteStatus($inviteStatus);
```
Comment (SurionA) [inline lib/JSON/Plugin.php]: So if `isset($jsonData->{'invite-reply'}->invitestatus) === false` we still call $node->updateInviteStatus($inviteStatus)  with `$invitestatus` variable not defined as you define it in the if block above

Should be with the first conditional : `isset($jsonData->{'invite-reply'}) && $jsonData->{'invite-reply'}->{'invitestatus'}`

File: tests/CalDAV/SharedCalendarTest.php [hunk 3]
```diff
@@ -343,6 +343,26 @@ function testGetInviteStatus() {
...
+        $backend = $this->getBackend();
+
+        $calendarESN = new \ESN\CalDAV\CalendarHome($backend, ['uri' => 'principals/user/54b64eadf6d7d8e41d263e0f']);
+        $sharedCalendarESN =  $calendarESN->getChild('events');
+
+        $status = 5;
+        $sharedCalendarESN->updateInviteStatus($status);
+
+        $acl = $sharedCalendarESN->getACL();
+        $isPresent = true;
```
Comment (SurionA) [inline tests/CalDAV/SharedCalendarTest.php]: So your test is always true ?

File: lib/CalDAV/SharedCalendar.php (hunk 1 above)
Comment (renaudboyer) [inline lib/CalDAV/SharedCalendar.php]: Code is done this way in Sabre

File: lib/JSON/Plugin.php [hunk 4]
```diff
@@ -800,6 +800,37 @@ function isHiddenPrivateEvent($vevent, $node) {
         return $vevent->CLASS == 'PRIVATE' && $node->getOwner() != $this->currentUser;
     }
 
+    function handleJsonRequest($path, $node, $jsonData) {
+        if (isset($jsonData->{'invite-reply'})) {
```
Comment (abenchi) [inline lib/JSON/Plugin.php]: and if we have nothing set, what will send the server??

File: lib/JSON/Plugin.php [hunk 5]
```diff
@@ -800,6 +800,37 @@ function isHiddenPrivateEvent($vevent, $node) {
...
+            return $this->updateInviteStatus($path, $node, $jsonData);
+        } else if (isset($jsonData->share)) {
+            return $this->updateSharees($path, $node, $jsonData);
+        }
+    }
+
+    function updateInviteStatus($path, $node, $jsonData) {
+        if (isset($jsonData->{'invite-reply'})) {
```
Comment (abenchi) [inline lib/JSON/Plugin.php]: we already checked for it!! line 804

File: lib/JSON/Plugin.php (hunk 4 above)
Comment (renaudboyer) [inline lib/JSON/Plugin.php]: Fixed with a 400 error

**Pull request merged**